### DISCUSSION TITLE
[UE] 区分 EditorOnly Properties / UFunctions

### DISCUSSION THREAD

**Discussion by BlurryLight** [Tencent/puerts#1800] [Ideas]
puer默认导出的时候不区分EditorOnly的Property/UFunction，有些客户端同学在runtime的场景下还在使用这些函数/变量，造成运行期的时候才出现报错。

需要处理的场景:
- 被WITH_EDITORONLY_DATA包裹的 UPROPERTY
- 被WITH_EDITOR包裹的 UFUNCTION  
- Editor的Module导出的所有UFunction / UProperty

处理不了的场景:
- 被WITH_EDITORONLY_DATA包裹的 UFUNCTION ( 可以生成合法的C++代码，但是UHT不会Parse这种情况，不会带上FUNC_EditorOnly)

目前的一个思路是在导出的时候，对这种带有Editor属性的，追加一个`_EditorOnly`的后缀在后面以强调这种属性。

**Comment by BlurryLight** [comment]
后面提个PR过来

**Comment by chexiongsheng** [comment]
河里
